### ISSUE TITLE
BibTex not working for GH Action build

### ISSUE THREAD

**Issue by thomasckng** [showyourwork/showyourwork#486]
My local build works normally, but the GH Action builds gives the following error and fails to produce any references. Does anyone know how to debug this? I am not sure where to add the print option.

Error: https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/actions/runs/11604085930/job/32312253716#step:4:413
> warning: errors were issued by BibTeX, but were ignored; use --print and/or --keep-logs for details.

GH generated pdf: https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/blob/7051e1d0786e92f9aeeee8dcb7e7826ae3450e2e/ms.pdf

**Comment by HealthyPear** [comment]
Hi @thomasckng sorry coming back to this since quite a while...(we are lacking maintainers!)

Do you still have this issue?

I took a look to your manuscript and I don't quite get where you are telling latex from where to grab your references.

Somewhere in your preamble you should have at least something like the following, am I correct?

```latex
\usepackage{biblatex}
\addbibresource{bib.bib}
```

**Comment by thomasckng** [comment]
@HealthyPear Thanks for the reply. Seems the `documentclass` I am using uses `natbib`. The following error shows up during the local build when I add `\usepackage{biblatex}`.
```
Package biblatex Error: Incompatible package 'natbib'.
...
The 'natbib' package and biblatex are incompatible.
```

The [current version](https://github.com/thomasckng/det-frame-cosmo-with-FIGARO) still works normally locally.

**Comment by HealthyPear** [comment]
> The following error shows up during the local build when I add \usepackage{biblatex}.

I think this is normal, you cannot use both at the same time

I managed to reproduce your manuscript locally, indeed I see the citations there, but in the remotely-produced PDF they do not work.

Can you see them on Overleaf?

**Comment by thomasckng** [comment]
Yes, overleaf works too.

**Comment by thomasckng** [comment]
> warning: errors were issued by BibTeX, but were ignored; use --print and/or --keep-logs for details.

Is there a way to print the error in the remote build?

**Comment by HealthyPear** [comment]
yes, you should edit your `showyourwork.yml` to add this flag to the arguments sent to _tectonic_ at the act of building the manuscript

`tectonic_args: ["--print"]`

**Comment by HealthyPear** [comment]
The 2 logs to compare are `.showyourwork/compile/ms.log` for the local one and the omologous one from the logs which you can download at the bottom of the workflow page
https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/actions/runs/12980248356

what I see is that in the remote there are a lot of natbib warnings which I do not see in the local log, e.g.

```
Package natbib Warning: Citation `Riess:2021jrx' on page 1 undefined on input l
ine 50.
```

so it seems that on the remote the bibliography is not found or that the natbib package cannot work as it should...

**Comment by HealthyPear** [comment]
I found another difference in the output file of (I think bibtex, or natbib) called `ms.blg`

on the remote one I see

```
This is BibTeX, Version 0.99d
Capacity: max_strings=35307, hash_size=35307, hash_prime=30011
The top-level auxiliary file: ms.aux
I couldn't open style file sn-APS.bst
---line 3 of file ms.aux
 : \bibstyle{sn-APS
 :                 }
I'm skipping whatever remains of this command
I found no style file---while reading file ms.aux
(There were 2 error messages)
```

while on the local one

```
This is BibTeX, Version 0.99d
Capacity: max_strings=35307, hash_size=35307, hash_prime=30011
The top-level auxiliary file: ms.aux
The style file: sn-APS.bst
Database file #1: bib.bib
Warning--entry type for "ng_2024_13968239" isn't style-file defined
--line 722 of file bib.bib
Warning--entry type for "ligo_scientific_collaboration_and_virgo_2022_6513631" isn't style-file defined
--line 736 of file bib.bib
Warning--entry type for "ligo_scientific_collaboration_and_virgo_2023_8177023" isn't style-file defined
--line 750 of file bib.bib
Warning--empty journal in Mukherjee:2022afz
Warning--empty journal in Marchant:2023wno
Warning--empty journal in Farah:2024xub
Warning--empty journal in MaganaHernandez:2024uty
Warning--empty journal in Li:2024rmi
Warning--empty journal in Lorenzo-Medina:2024opt
(There were 9 warnings)
```

so it seems that on the remote the file `sn-APS.bst` is not used or not recognized.

I see at least 2 ways to try to fix this (still, I cannot explain why it works locally and not remotely):

- add explicitly `\bibliographystyle{sn-APS}` to your manuscript after `\usepackage{natbib}`
- add "src/tex/sn-APS.bst" to the dependencies of the manuscript together with the bib file

**Comment by HealthyPear** [comment]
maybe try those 2 solutions separately, so we can pinpoint better what might be the issue

**Comment by thomasckng** [comment]
For adding `\bibliographystyle{sn-APS}`, it gives the following error even for local build: `Illegal, another \bibstyle command : \bibstyle : {sn-APS}`.

For adding `src/tex/sn-APS.bst` to the dependencies, I think it gives the same error as before.
https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/actions/runs/12990333380/job/36225257468

**Comment by HealthyPear** [comment]
> For adding `\bibliographystyle{sn-APS}`, it gives the following error even for local build: `Illegal, another \bibstyle command : \bibstyle : {sn-APS}`.

This checks out, you have it already in the class definition

https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/blob/fc94c6c470af6e507b29f5673d42a0e4f0dd56e1/src/tex/sn-jnl.cls#L1671

a possible doubt is that the class looks for `sn-APS` and you file is called `sn-aps.bst`

not sure if this command is case-sensitive, but again, no idea why this should work locally and not remotely

though in the .blg file I see explicitly `I couldn't open style file sn-APS.bst`

so maybe rename the file either in the class definition or in your repository tree and see what happens

> For adding `src/tex/sn-APS.bst` to the dependencies, I think it gives the same error as before. https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/actions/runs/12990333380/job/36225257468

this does not hurt anyway :)

**Comment by HealthyPear** [comment]
I am more and more convinced that this is a matter of file-system case-(in)sensitivity.

The GitHub runner uses Ubuntu 24.04.1 LST on which the default file system is typically ext4, which is case-sensitive.
On my local machine (macos) I have APFS, which is case-INsensitive. Is this the case for your local machine?

Can you try to check with e.g. `mount | grep "on / "`?

**Comment by thomasckng** [comment]
Changing [the line you mentioned](https://github.com/thomasckng/det-frame-cosmo-with-FIGARO/blob/2ef2cb9cc5a8aceae2c1624c0f18459d4d1eb4c8/src/tex/sn-jnl.cls#L1671) fixes the problem! Thanks a lot.

For `mount | grep "on / "`, it outputs `/dev/disk3s1s1 on / (apfs, sealed, local, read-only, journaled)` locally.

**Comment by HealthyPear** [comment]
Great! So that was the problem indeed...

@afm I wonder what is the best way to go on our side. This is not really a showyourwork issue, but not doing anything will also not help anybody. Perhaps we should add a note section in the documentation about Latex?
